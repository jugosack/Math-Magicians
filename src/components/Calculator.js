import './Calculator.scss';

function Calculator() {
  return (
    <h1 className="header">Hello World</h1>
  );
}
export default Calculator;
