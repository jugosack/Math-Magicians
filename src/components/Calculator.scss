.space {
  margin-top: 50px;
}

.container {
  margin: auto;
  width: 30%;
  height: 70%;
  padding: 10px;
  border: 1px solid #f2f2f2;
  -webkit-box-shadow: 1px 1px 4px 0 rgba(176, 176, 176, 1);
  -moz-box-shadow: 1px 1px 4px 0 rgba(176, 176, 176, 1);
  box-shadow: 1px 1px 4px 0 rgba(176, 176, 176, 1);
  background-color: #f8f8f8;
}

.result1 {
  height: 40px;
  background-color: gray;

  /*   padding: 10px; */
  color: white;
  text-align: right;
  font-size: 20px;
}

table {
  width: 100%;
}

td {
  width: 25%;
}

.result {
  font-size: 24px;
  padding: 16px 5px;
  font-weight: 700;
  text-align: right;
  width: 96%;
  border: none;
  background-color: #f8f8f8;
}

.global {
  // padding: 15px;
  font-weight: bold;
  font-size: 16px;
  border: none;
  border-radius: 3px;
  background-color: #f2f2f2;
  width: 100%;
  height: 70px;
}

/* the global class padding reset for first-row class button */
.first-row {
  padding: 7px;
}

.grey {
  background-color: #f2f2f2;
}

#plus,
#slash,
#ex,
#minus,
#equal {
  background-color: bisque;
}
