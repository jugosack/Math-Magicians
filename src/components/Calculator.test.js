import { render, screen } from '@testing-library/react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

describe('snapshot testing', () => {
  it('first snapshot', () => {
    const itemRender = renderer.create(<Calculator />).toJSON();
    expect(itemRender).toMatchSnapshot();
  });
});

describe('Calculator item', () => {
  it('testing item', () => {
    render(<Calculator />);
    const ele = screen.getByTestId('calculatorTest');
    expect(ele).toBeInTheDocument();

    const td0 = screen.getByDisplayValue('AC');
    expect(td0).toHaveDisplayValue('AC');

    const td1 = screen.getByDisplayValue('+/-');
    expect(td1).toHaveDisplayValue('+/-');

    const td2 = screen.getByDisplayValue('+');
    expect(td2).toHaveDisplayValue('+');
  });
});

describe('Rendering Buttons in Calculator', () => {
  it('rendering correct button', () => {
    const { getAllByDisplayValue } = render(<Calculator />);
    getAllByDisplayValue('AC');
    getAllByDisplayValue('+');
    getAllByDisplayValue('-');
    getAllByDisplayValue('x');
    getAllByDisplayValue('1');
    getAllByDisplayValue('2');
    getAllByDisplayValue('3');
    getAllByDisplayValue('4');
    
    /// /Add Test By John ////
  });
});
