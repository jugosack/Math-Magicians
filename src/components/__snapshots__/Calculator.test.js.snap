// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing first snapshot 1`] = `
Array [
  <div
    className="space"
  />,
  <div
    className="container"
    data-testid="calculatorTest"
  >
    <table>
      <tbody>
        <tr
          className="result1"
        >
          <td
            colSpan={4}
          >
            0
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="global first-row"
              onClick={[Function]}
              type="button"
              value="AC"
            />
          </td>
          <td>
            <input
              className="global first-row"
              onClick={[Function]}
              type="button"
              value="+/-"
            />
          </td>
          <td>
            <input
              className="global first-row"
              onClick={[Function]}
              type="button"
              value="%"
            />
          </td>
          <td>
            <input
              className="global first-row"
              id="slash"
              onClick={[Function]}
              type="button"
              value="รท"
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="7"
            />
          </td>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="8"
            />
          </td>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="9"
            />
          </td>
          <td>
            <input
              className="global"
              id="ex"
              onClick={[Function]}
              type="button"
              value="x"
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="4"
            />
          </td>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="5"
            />
          </td>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="6"
            />
          </td>
          <td>
            <input
              className="global"
              id="minus"
              onClick={[Function]}
              type="button"
              value="-"
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="1"
            />
          </td>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="2"
            />
          </td>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="3"
            />
          </td>
          <td>
            <input
              className="global"
              id="plus"
              onClick={[Function]}
              type="button"
              value="+"
            />
          </td>
        </tr>
        <tr>
          <td
            colSpan="2"
          >
            <input
              className="global grey"
              onClick={[Function]}
              type="button"
              value="0"
            />
          </td>
          <td>
            <input
              className="global"
              onClick={[Function]}
              type="button"
              value="."
            />
          </td>
          <td>
            <input
              className="global"
              id="equal"
              onClick={[Function]}
              type="button"
              value="="
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
